adlfs/core.py

```diff
@@ -743,6 +743,9 @@ def _put_data(rest, op, path, data, max_attempts=10, **kwargs):
         except (PermissionError, FileNotFoundError):
             raise
         except Exception as e:
+            if '"BadOffsetException"' in repr(e):
```

Comment:
(but we could rename the `loc` attribute to `_loc` to indicate that users shouldn't mess with it)